src/Illuminate/Support/Arr.php

```diff
@@ -290,7 +290,8 @@ public static function has($array, $key)
         }
 
         foreach (explode('.', $key) as $segment) {
-            if (! is_array($array) || ! array_key_exists($segment, $array)) {
+            if ((! is_array($array) || ! array_key_exists($segment, $array)) &&
+                (! $array instanceof ArrayAccess || ! $array->offsetExists($segment))) {
```

Comment:
No it's ok, because of the short-circuit, `offsetExists` is executed iff instance of `ArrayAccess`. However I agree this conditional isn't intuitive to read.